### DISCUSSION TITLE
Nested routes

### DISCUSSION THREAD

**Discussion by DenisKnezBeforeNine** [remix-run/remix#9814] [Q&A] [answered]
I have 2 route files in my project

`company-ad.$id.project.$project-id`     which I want to access with  /company-ad/11/project/2
`company-ad.$id`                                     and the other route which I want to access with /company-ad/13

Both of these routes work on their own but if I have both of them then only the /company-ad/13 works. When I go to the /company-ad/11/project/2 it open the  /company-ad/13, the url is still /company-ad/11/project/2 but the page contents is of the /company-ad/13. But then when I remove the **company-ad.$id**  file the route with the project works fine.

What I assume the reason for this is because it get the one it matches first. But is this the intended logic? 
How can I achieve the routes that I want?

**Comment by alexanderson1993** [comment, marked as answer]
`company-ad.$id` is actually a layout route, not a leaf route. In order for any sub-routes to render their components, you either need to
- render `<Outlet>` somewhere inside `company-ad.$id`, which is where the leaf route will render
- Change the name of `company-ad.$id` to `company-ad.$id._index` so it is no longer treated as a layout route.
